src/Build/Evaluation/Context/EvaluationContext.cs

```diff
@@ -48,7 +48,7 @@ public enum SharingPolicy
         /// <summary>
         /// Key to file entry list. Example usages: cache glob expansion and intermediary directory expansions during glob expansion.
         /// </summary>
-        private ConcurrentDictionary<string, ImmutableArray<string>> FileEntryExpansionCache { get; }
+        private ConcurrentDictionary<string, IReadOnlyList<string>> FileEntryExpansionCache { get; }
```

Comment:
very minor: could this be readonly as it's instantiated in the .ctor?
However, readonly gives the impression of immutability after instantiation, not the case for collections.

Reply:
It's a property and defining only `get;` actually makes it read-only. The `readonly` keyword would not be valid here.